src/main/java/io/odpf/firehose/sinkdecorator/SinkFinal.java

```diff
@@ -0,0 +1,35 @@
+package io.odpf.firehose.sinkdecorator;
```

Comment:
we can just capture the global metric in abstract sink itself, the current responsibility i believe is not good enough to be a seperate class.

Reply:
Abstract sink will not have info about ignored messages. It's the bottom most call.. we need to check the messages when everything else is processed.

Reply:
consumer-> sinkfinal -> dlq -> retry -> abstractsink
                                                                                 |
            consumer<-sinkfinal <-dlq<-retry<----  V